### PR TITLE
Adding model registry URI and tracking to predict step

### PR DESCRIPTION
**Pull request by jerrylian-db**
## Related Issues/PRs

#xxx

## What changes are proposed in this pull request?

This PR makes the Batch Scoring work end to end.

## How is this patch tested?

- [ ] I have written tests (not required for typo or doc fix) and confirmed the proposed feature/bug-fix/change works.

Automated tests are blocked by needing to update of the mlp regression example submodule in the MLflow repository.

I tested this manually. Steps:
* Downloaded the latest [MLP regression template](https://github.com/mlflow/mlp-regression-template)
* Created a new Python 3.8.13 virtual environment
* Installed the template requirements
* Installed the mlflow branch I'm working on
* Ran the register step and then ran the predict step. Result below:
Screen Shot 2022-09-28 at 10 29 25 AM

## Does this PR change the documentation?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly by following the steps below.

1. Click the `Details` link on the `Preview docs` check.
2. Find the changed pages / sections and make sure they render correctly.

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/pipelines`: Pipelines, Pipeline APIs, Pipeline configs, Pipeline Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface
- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: mlflow/utils/_spark_utils.py [hunk 1]
```diff
@@ -32,6 +32,7 @@ def _create_local_spark_session_for_pipelines():
         .config(
             "spark.sql.catalog.spark_catalog", "org.apache.spark.sql.delta.catalog.DeltaCatalog"
         )
+        .config("spark.sql.execution.arrow.pyspark.enabled", "true")
```
Comment (jinzhang21) [inline mlflow/utils/_spark_utils.py:35]: Is this change intentional?

**Review by jinzhang21 — COMMENTED** [review]
Can we add a test to test_predict.py to verify the changes work?

File: mlflow/utils/_spark_utils.py (hunk 1 above)
Comment (jerrylian-db) [reply mlflow/utils/_spark_utils.py:35]: Yes, this change is necessary. Why? The predict step reads data from the ingest step by first reading it with pandas and then converting the pandas DF into a spark DF. We do not directly read with spark because spark cannot read directly from the driver disk. However, in this conversion process, without "arrow" being enabled, the spark DF will change the datatypes of the pandas DF in a way that breaks the input schema of the registered model. Hence, I added this configuration to spark.

**Review by jinzhang21 — APPROVED** [review]
